src/clients/save_network.rb

```diff
@@ -263,6 +266,63 @@ def configure_target
       nil
     end
 
+    # Applies part of AY configuration at the end of first stage
+    #
+    # Intended mainly for steps which cannot be done in AY's second stage
+    #
+    # FIXME: Currently used only for applying udev rules during network
+    # installations (ssh, vnc, ...). It was introduced as a quick fix for
+    # bnc#944349, so it is currently limited only on {ssh|vnc} installations.
+    # Once properly analyzed and tested then starting of whole network second
+    # stage can be moved here.
+    def ay_mode_configuration
+      # TODO: run only when in {ssh|vnc} mode of AY
+      return if !Mode.autoinst
+      return if !(Linuxrc.usessh || Linuxrc.vnc)
+
+      ay_profile = Profile.current
+
+      log.info("Applying udev rules according AY profile")
+
+      return if ay_profile.nil? || ay_profile.empty?
+      return if ay_profile["networking"].nil? || ay_profile["networking"].empty?
+
+      udev_rules = ay_profile["networking"]["net-udev"]
+      log.info("- udev rules: #{udev_rules}")
+
+      return if udev_rules.nil? || udev_rules.empty?
+
+      LanItems.Read
+
+      udev_rules.each do |rule|
+        name_to = rule["name"]
+        attr = rule["rule"]
+        key = rule["value"].downcase
+        # currently we're interrested only on those interfaces which are already
+        # configured - such interfaces cannot be restarted during second stage
+        matching_item = LanItems.Items.find { |_, i| i["hwinfo"]["busid"].downcase == key || i["hwinfo"]["mac"].downcase == key }
```

Comment:
hint is to use `index, matching_item = ...` this way below you do not need to use nasty `matching_item[1]`